### ISSUE TITLE
extension cashes on Debian aarch64 with syntaxDiagnosticsForFile is not iterable

### ISSUE THREAD

**Issue by drue** [rescript-lang/rescript-vscode#794]
On Debain Sid, Thinkpad X13s (Snapdragon 8cx / aarch64).  I didn't have any trouble building the extension using the instructions in CONTRIBUTING.md , Dune and Ocaml seem to be working fine, but the results were the same when loading the generated .vsix directly.

```
/home/drue/.vscode-oss/extensions/chenglou92.rescript-vscode-1.18.0/server/out/server.js:665
            diagnostics: [...syntaxDiagnosticsForFile, ...compilerDiagnosticsForFile],
                             ^

TypeError: syntaxDiagnosticsForFile is not iterable
    at updateDiagnosticSyntax (/home/drue/.vscode-oss/extensions/chenglou92.rescript-vscode-1.18.0/server/out/server.js:665:30)
    at process.onMessage (/home/drue/.vscode-oss/extensions/chenglou92.rescript-vscode-1.18.0/server/out/server.js:821:13)
    at process.emit (node:events:513:28)
    at emit (node:internal/child_process:939:14)
    at process.processTicksAndRejections (node:internal/process/task_queues:84:21)
[Info  - 6:30:43 AM] Connection to server got closed. Server will restart.
true
```

**Comment by drue** [comment]
I should say that I got the same result with a fresh checkout of rescript-proect-template and a fresh install of the extension from the marketplace.

**Comment by drue** [comment]
Thanks for the tip!  I got it working.  For anyone else wanting to use the Rescript VSCode Extension on Linux ARM64/AARCH64 before it is officially supported, I used the following workaround:

1. install opam, dune, and cppo  (was packaged in Ubuntu)
2. cd into the analysis directory in the extension source and "make" to build rescript-editor-analysis.exe binary
3. download or build a recent version of the extension, version >= 19
4. open the .vsix file in a ZIP archive editor and copy the rescript-editor-analysis.exe binary  into a new directory "linuxarm64" inside the existing "extension/server/analysis_binaries"  directory
5. once you have put the analysis binary inside the vsix in extension/server/analysis_binaries/linuxarm64/rescript-editor-analysis.exe install the visx file into vs code by going into the extensions pane , clicking the ellipses menu, and choosing "Install from VSIX..."

happy hacking!
